src/main.rs

```diff
@@ -52,6 +62,22 @@ fn main() {
         }),
         ..Default::default()
     };
+
+    let speed_mode: usize = matches
+        .value_of("SPEED_MODE")
+        .map(|val| val.parse().expect("Speed mode must be in range [0; 2]"))
+        .unwrap();
+
+    if matches.is_present("SCENE_CHANGE_DETECTION_SPEED") {
```

Comment:
I think the name of this argument is incorrect, and should be `SPEED_MODE`.

I'd actually go a little further in cleaning it up to be something like this (from lines 66-79):

```rust
if let Some(speed_mode) = matches.value_of("SPEED_MODE") {
    let speed_mode: usize = speed_mode.parse().expect("Speed mode must be in range [0; 2]");
    opts.fast_analysis = if speed_mode == 0 {
        SceneDetectionSpeed::Slow
    } else if speed_mode == 1 {
        SceneDetectionSpeed::Medium
    } else if speed_mode == 2 {
        SceneDetectionSpeed::Fast
    } else {
        panic!("Speed mode must be in range [0; 2]");
    };
}
```